buffother.lic

```diff
@@ -0,0 +1,47 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#buffother
+=end
+custom_require.call(%w(common common-arcana events spellmonitor drinfomon common-travel))
+class Buffother
+  include DRC
+  include DRCA
+  include DRCT
+
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'recipient', regex: /\w+/, description: 'Person to buff' },
+        { name: 'waggle', regex: /\w+/, description: 'Waggle set to cast' },
+        { name: 'room', regex: /\d+/, optional: true, description: 'Room to move to' }
+      ]
+    ]
+    args = parse_args(arg_definitions)
+    @settings = get_settings
+    @recipient = args.recipient.capitalize
+    @waggle = args.waggle
+    @room = args.room
+    move_to_room
+  end
+
+  def move_to_room
+    echo "Moving to room"
+    DRCT.walk_to(@room)
+    cast_buff_spells
+    pause 1
+  end
+
+  def cast_buff_spells
+    echo "In cast_buff_spells"
+    data = @settings.waggle_sets["#{@waggle}"].values.each do |data|
+      data['cast'] = "cast #{@recipient}"
+      DRCA.do_buffs(@settings, @waggle)
+    end
+    cleanup
+  end
+
+  def cleanup
+    fput("release mana")
+  end
```

Comment:
To ensure this happens even if the script exits prematurely, either on purpose (user stopping script) or by some other reason, I recommend adding a [before_dying](https://elanthipedia.play.net/Lich_scripting_reference#before_dying) method at the end of this. class that calls your `cleanup` method.

```
before_dying do
    cleanup
end
```

Additionally, since this ensures your cleanup method is called on script exit, you could remove line 39 as it would be redundant to the failsafe invocation of it.

Reply:
Love it. Change made.